server/src/main/java/org/apache/iotdb/db/rescon/SystemInfo.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.rescon;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.PriorityQueue;
+import org.apache.iotdb.db.conf.IoTDBConfig;
+import org.apache.iotdb.db.conf.IoTDBDescriptor;
+import org.apache.iotdb.db.engine.flush.FlushManager;
+import org.apache.iotdb.db.engine.storagegroup.StorageGroupInfo;
+import org.apache.iotdb.db.engine.storagegroup.TsFileProcessor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SystemInfo {
+
+  private static final IoTDBConfig config = IoTDBDescriptor.getInstance().getConfig();
+  private static final Logger logger = LoggerFactory.getLogger(SystemInfo.class);
+
+  private long totalSgMemCost;
+  private volatile boolean rejected = false;
+
+  private Map<StorageGroupInfo, Long> reportedSgMemCostMap = new HashMap<>();
+
+  private static final double FLUSH_PROPORTION = config.getFlushProportion();
+  private static final double REJECT_PROPORTION = config.getRejectProportion();
+
+  /**
+   * Report current mem cost of storage group to system.
+   *
+   * @param storageGroupInfo storage group
+   */
+  public synchronized void reportStorageGroupStatus(StorageGroupInfo storageGroupInfo) {
+    long delta = storageGroupInfo.getSgMemCost() -
+        reportedSgMemCostMap.getOrDefault(storageGroupInfo, 0L);
+    totalSgMemCost += delta;
+    logger.debug("Report Storage Group Status to system. "
+          + "Current sg mem cost is {}, delta is {}.", totalSgMemCost, delta);
+    reportedSgMemCostMap.put(storageGroupInfo, storageGroupInfo.getSgMemCost());
+    storageGroupInfo.setLastReportedSize(storageGroupInfo.getSgMemCost());
+    if (getTotalMemCost() >= config.getAllocateMemoryForWrite() * FLUSH_PROPORTION) {
+      logger.debug("The total storage group mem costs are too large, call for flushing. "
+          + "Current sg cost is {}", totalSgMemCost);
+      flush();
+    }
+    if (getTotalMemCost() >= config.getAllocateMemoryForWrite() * REJECT_PROPORTION) {
+      logger.debug("Change system to reject status...");
```

Comment:
```suggestion
      logger.info("Change system to reject status...");
```
Add current memory info